RELEASES.md

```diff
@@ -1,3 +1,96 @@
+Version 1.35.0 (2019-05-23)
+==========================
+
+Language
+--------
+- [`FnOnce`, `FnMut`, and the `Fn` traits are now implemented for `Box<FnOnce>`,
```

Comment:
I'm debating whether we should expand this bullet with "why should you care" information... On the one hand, it is probably the most important change this release. On the other, the bullet has all information that is sufficient from a "this is what happened" POV.

Reply:
This is something that could be expanded on in the blog post.